extensions/binding/org.eclipse.smarthome.binding.sonos/src/main/java/org/eclipse/smarthome/binding/sonos/handler/ZonePlayerHandler.java

```diff
@@ -450,43 +509,88 @@ public void onValueReceived(String variable, String value, String service) {
+        return oldValue != null ? !oldValue.equals(value) : value != null;
+    }
+
+    /**
+     * Similar to the AVTransport eventing, the Queue events its state variables
+     * as sub values within a synthesized LastChange state variable.
+     */
+    private boolean isQueueEvent(String variable) {
+        return "LastChange".equals(variable);
+    }
+
+    private void updateGroupCoordinator() {
+        updateState(new ChannelUID(getThing().getThingTypeUID(), getThing().getUID(), COORDINATOR),
+                new StringType(getCoordinator()));
+        coordinatorHandler = getHandlerByName(getCoordinator());
+    }
+
+    private void removeSubscription() {
+        synchronized (upnpLock) {
+            // Set up GENA Subscriptions
+            if (service.isRegistered(this)) {
+                for (String subscription : SERVICE_SUBSCRIPTIONS) {
+                    service.removeSubscription(this, subscription);
+                }
+                service.unregisterParticipant(this);
             }
-            service.unregisterParticipant(this);
         }
     }
 
-    private synchronized void onUpdate() {
-        if (service.isRegistered(this)) {
-            if (pollingJob == null || pollingJob.isCancelled()) {
-                Configuration config = getThing().getConfiguration();
-                // use default if not specified
-                int refreshInterval = DEFAULT_REFRESH_INTERVAL;
-                Object refreshConfig = config.get("refresh");
-                if (refreshConfig != null) {
-                    refreshInterval = ((BigDecimal) refreshConfig).intValue();
+    private void onUpdate() {
+        synchronized (upnpLock) {
+            if (service.isRegistered(this)) {
+                for (String subscription : SERVICE_SUBSCRIPTIONS) {
+                    service.addSubscription(this, subscription, SUBSCRIPTION_DURATION);
+                }
+
+                if (pollingJob == null || pollingJob.isCancelled()) {
+                    Configuration config = getThing().getConfiguration();
+                    // use default if not specified
+                    int refreshInterval = DEFAULT_REFRESH_INTERVAL;
+                    Object refreshConfig = config.get("refresh");
+                    if (refreshConfig != null) {
+                        refreshInterval = ((BigDecimal) refreshConfig).intValue();
+                    }
+                    pollingJob = scheduler.scheduleAtFixedRate(pollingRunnable, 0, refreshInterval, TimeUnit.SECONDS);
                 }
-                pollingJob = scheduler.scheduleAtFixedRate(pollingRunnable, 0, refreshInterval, TimeUnit.SECONDS);
             }
         }
     }
 
+    private void updatePlayerState() {
+        Map<String, String> result = service.invokeAction(this, "DeviceProperties", "GetZoneInfo", null);
+        if (result.isEmpty()) {
+            logger.debug("Sonos player " + getThing().getProperties().get(Thing.PROPERTY_MODEL_ID)
```

Comment:
I will try to figure out a way to update the Sonos thing state without invoking upnp calls.